mri/reconstructors/base.py

```diff
@@ -0,0 +1,175 @@
+# -*- coding: utf-8 -*-
+##########################################################################
+# pySAP - Copyright (C) CEA, 2017 - 2018
+# Distributed under the terms of the CeCILL-B license, as published by
+# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
+# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
+# for details.
+##########################################################################
+
+# System import
+import warnings
+
+# Package import
+from ..operators.linear.wavelet import WaveletUD2, WaveletN
+from ..optimizers import pogm, condatvu, fista
+from ..optimizers.utils.cost import GenericCost
+
+# Third party import
+from modopt.opt.proximity import SparseThreshold
+from modopt.opt.linear import Identity
+
+
+class ReconstructorBase(object):
+    """ This is the base reconstructor class for reconstruction.
+    This class holds some parameters that is common for all MR Image
+    reconstructors
+    Parameters
+    ----------
+    fourier_op: object of class FFT, NonCartesianFFT or Stacked3DNFFT in
+                mri.operators
+        Defines the fourier operator F.
+    linear_op: object
+        Defines the linear sparsifying operator Wt. This must operate on x and
+        have 2 functions, op(x) and adj_op(coeff) which implements the
+        operator and adjoint operator. For wavelets, this can be object of
+        class WaveletN or WaveletUD2 from mri.operators .
+    prox_op: object that implements proximal operator
+        Defines the proximity operator for the regularization function.
+        For example, for L1 Norm, the proximity operator is Thresholding
+    mu: float
+        If prox_op is None, the value of mu is used to form a proximity
+        operator that is soft thresholding of the wavelet coefficients.
```

Comment:
Explicitly add that if `prox_op` is defined, this is ignored (same in other docs).